nautobot/core/api/filter_backends.py

```diff
@@ -0,0 +1,29 @@
+from django_filters.rest_framework.backends import DjangoFilterBackend
+
+
+class NautobotFilterBackend(DjangoFilterBackend):
+    """Custom filtering backend for use with django-rest-framework and django-filters."""
+
+    def get_filterset_kwargs(self, request, queryset, view):
+        """
+        Get the kwargs that should be passed through when constructing a FilterSet corresponding to a given request.
+
+        This extends the base DjangoFilterBackend method to explicitly exclude query parameters that we know to be
+        non-filterset parameters.
+        """
+        kwargs = super().get_filterset_kwargs(request, queryset, view)
+        # The default 'data' is a reference to request.GET, which is immutable; copies of this data are mutable
+        data = kwargs["data"].copy()
+
+        for non_filter_param in (
+            "api_version",  # used to select the Nautobot API version
+            "brief",  # used to select NestedSerializer rather than Serializer
+            "format",  # "json" or "api", used in the interactive HTML REST API views
+            "include",  # used to include computed fields (excluded by default)
+            "limit",  # pagination
+            "offset",  # pagination
+        ):
+            data.pop(non_filter_param, None)
```

Comment:
If I am understanding this correctly this is just removing the "approved" non-filter parameters.

Since the goal of this PR is to reject unrecognized filter parameters, we would still be passing down parameters that aren't filter parameters (as intended for the validation/rejection).

I can't think of a better phrasing/naming convention but something along the lines of `control_params` or `view_params` might make this more intuitively grok-able.